src/governance/CoveTokenBazaarAuction.sol

```diff
@@ -0,0 +1,154 @@
+ * - Senders: Bazaar Uncapped Auction Factory for creating the auction, Bazaar Uncapped Auction for users to claim
+ * tokens
+ * - Receivers: None
+ * It inherits from OpenZeppelin's ERC20 and Ownable contracts.
+ */
+contract CoveTokenBazaarAuction is ERC20, Ownable2Step {
+    /// @notice Mapping to track addresses allowed to receive transfers.
+    mapping(address => bool) public allowedReceiver;
+    /// @notice Mapping to track addresses allowed to initiate transfers.
+    mapping(address => bool) public allowedSender;
+    /// @notice State variable to make the events orderable for external observers if they are called in the same block.
+    uint256 private _eventId;
+    /// @notice Total supply of the token.
+    uint256 private constant _TOTAL_SUPPLY = 95_000_000 ether;
+    /// @notice Address of the Bazaar Uncapped Auction Factory contract.
+    /// https://etherscan.io/address/0x47612eabFbE65329AeD1ab1BF3FCbAE493aEf460
+    address private constant _BAZAAR_UNCAPPED_AUCTION_FACTORY = 0x47612eabFbE65329AeD1ab1BF3FCbAE493aEf460;
+
+    /**
+     * @notice Emitted when an address is granted permission to initiate transfers.
+     * @param target The address that is being allowed to send tokens.
+     * @param eventId An identifier for the event to order events within the same block.
+     */
+    event SenderAllowed(address indexed target, uint256 eventId);
+    /**
+     * @notice Emitted when an address has its permission to initiate transfers revoked.
+     * @param target The address that is being disallowed from sending tokens.
+     * @param eventId An identifier for the event to order events within the same block.
+     */
+    event SenderDisallowed(address indexed target, uint256 eventId);
+    /**
+     * @notice Emitted when an address is granted permission to receive transfers.
+     * @param target The address that is being allowed to receive tokens.
+     * @param eventId An identifier for the event to order events within the same block.
+     */
+    event ReceiverAllowed(address indexed target, uint256 eventId);
+    /**
+     * @notice Emitted when an address has its permission to receive transfers revoked.
+     * @param target The address that is being disallowed from receiving tokens.
+     * @param eventId An identifier for the event to order events within the same block.
+     */
+    event ReceiverDisallowed(address indexed target, uint256 eventId);
+
+    /**
+     * @notice Deploys this contract with the initial owner and minting the total supply to the owner.
+     * @param owner_ The address of the initial owner.
+     */
+    // slither-disable-next-line locked-ether
+    constructor(address owner_) payable Ownable() ERC20("Cove DAO Bazaar Auction Token", "COVE-BAZAAR") {
+        // Checks
+        if (owner_ == address(0)) {
+            revert Errors.ZeroAddress();
+        }
+        _transferOwnership(owner_); // Transfer ownership to the provided address
```

Comment:
does this mean the community multisig will also have to accept ownership before allowlisting the created bazaar auction contract?

Reply:
Nope. Only the external calls are 2 step. Unlikely we will take advantage of the 2step ownership transfers for this token but decided to side with the best practices.